torchrl/objectives/value/functional.py

```diff
@@ -332,6 +332,8 @@ def vec_generalized_advantage_estimate(
             lmbda=lmbda,
         )
 
+    not_terminated = (~terminated).to(dtype)
```

Comment:
I can't comment outside of the diff but ~15 lines below `not_terminated` is redefined. Maybe worth removing it as part of this PR.